numbers/integrate.go

```diff
@@ -6,6 +6,27 @@ import (
 	//"fmt"
 )
 
+//logIntegrate evaluates log(int_a^b f(x)dx) in cases where f returns log(f(x)). Uses the rectangle rule.
+func logIntegrate(f func(float64) float64, a float64, b float64, n int) float64 {
+	if a >= b {
+		log.Fatalf("logIntegrate failed, left bound must be smaller than right bound.")
+	}
+	var deltaX float64 = (b - a) / float64(n)
+	var logDeltaX float64 = math.Log(deltaX)
+	var currLeft float64 = a //this variable stores the left bound of the current rectangle.
+	var currRight float64 = a + deltaX
+	var answer float64
+	//first time, sets answer as the area of the first rectangle
+	answer = MultiplyLog(MidpointLog(f(currLeft), f(currRight)), logDeltaX)
+
+	for i := 1; i < n; i++ {
+		currLeft += deltaX
+		currRight += deltaX
+		answer += MultiplyLog(MidpointLog(f(currLeft), f(currRight)), logDeltaX)
```

Comment:
Not an error, but a chance to get a speed up.  I think you are evaluating f(x) twice as much as you need to because currRight of this time through the loop is currLeft of the next time through the loop.  If you do an initial evaluation of f(currLeft) you can then go through the loop only evaluating f(currRight) each time, then setting your variable for f(currLeft) to be f(currRight), so it is ready for the next time through the loop